### ISSUE TITLE
flutter pub get failing with More 3.7.0 and Flutter 3.0.2

### ISSUE THREAD

**Issue by ncuillery** [renggli/dart-more#11]
- Run `flutter create hello_world`
- Add `more: ^3.7.0` to the `pubspec.yaml` file
- Run `flutter pub get`

Got the following error:

```
$ flutter pub get 
Because every version of flutter_test from sdk depends on meta 1.7.0 and more >=3.6.0 depends on meta ^1.8.0, flutter_test from sdk is incompatible with more >=3.6.0.
So, because hello_world depends on both more ^3.7.0 and flutter_test from sdk, version solving failed.
Running "flutter pub get" in hello_world...                             
pub get failed (1; So, because hello_world depends on both more ^3.7.0 and flutter_test from sdk, version solving failed.)
```

**Comment by renggli** [comment]
Thank you for the report. Unfortunately there are always some packages that depend on older versions of common packages.  I suggest you report this to those packages, and in the meantime depend on an older version of `more` that is compatible with your dependencies?

**Comment by ncuillery** [comment]
In that case, the other package is `flutter_test` which dictates a fixed version of `meta`: https://github.com/flutter/flutter/blob/3.0.2/packages/flutter_test/pubspec.yaml#L39

They have already updated `meta` to 1.8.0 in master so at some points in the future, the incompatibility will be fixed by simply upgrading Flutter, but in the meantime this issue will block people who want to use the latest version of More.

It sucks for package developers, but I think the best practice is to keep `meta` at the lowest version possible. See few examples:
- https://github.com/felangel/bloc/blob/8ad0a14b4eb4dde97f62d04dece8e0a72f15d5f4/packages/bloc/pubspec.yaml#L13
- https://github.com/firebase/flutterfire/blob/ea03c6b349ccd63317f96e37c85daf7e11f2a28a/packages/firebase_messaging/firebase_messaging/pubspec.yaml#L22

**Comment by renggli** [comment]
I've published `more 3.7.1` that depends on `meta 1.7.0` instead.
